### PR TITLE
Challenges Admin

### PR DESCRIPTION
**Pull request by Leusev**
### Purpose

Challenges's component Admin section for Challenges crud

### CODE REVIEW

File: app/assets/config/decidim_challenges_manifest.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+//= link decidim/challenges/add_challenge.js
+//= link decidim/challenges/admin/challenges_form.js
+//= link_tree ../images/decidim
```
Comment (tramuntanal) [inline app/assets/config/decidim_challenges_manifest.js:3]: are we requiring Javsacripts from the images/ dir?
```suggestion
```

File: app/assets/javascripts/decidim/challenges/add_challenge.js.es6 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+$(() => {
+  const { attachGeocoding } = window.Decidim;
```
Comment (tramuntanal) [inline app/assets/javascripts/decidim/challenges/add_challenge.js.es6:2]: Do we need this line?

File: app/assets/javascripts/decidim/challenges/admin/challenges_form.js.es6 [hunk 3]
```diff
@@ -0,0 +1,4 @@
+$(() => {
+  const { attachGeocoding } = window.Decidim;
```
Comment (tramuntanal) [inline app/assets/javascripts/decidim/challenges/admin/challenges_form.js.es6:2]: Do we need this line?

File: app/controllers/decidim/challenges/admin/challenges_controller.rb.bak [hunk 4]
```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
```
Comment (tramuntanal) [inline app/controllers/decidim/challenges/admin/challenges_controller.rb.bak:1]: Remove this file please

File: app/events/decidim/challenges/publish_challenge_event.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# frozen-string_literal: true
+
+module Decidim
+  module Challenges
+    class PublishChallengeEvent < Decidim::Events::SimpleEvent
```
Comment (tramuntanal) [inline app/events/decidim/challenges/publish_challenge_event.rb:5]: Is this event being sent somewhere? shouldn't we remove it and the corresponding translations?

File: app/permissions/decidim/challenges/admin/permissions.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+module Decidim
+  module Challenges
+    module Admin
+      class Permissions < Decidim::DefaultPermissions
+        def permissions
+          # The public part needs to be implemented yet
+          return permission_action if permission_action.scope != :admin
+
+          allow! if permission_action.subject == :challenges && permission_action.action == :read && read_permission_action?
```
Comment (tramuntanal) [inline app/permissions/decidim/challenges/admin/permissions.rb:11]: why `&& permission_action.action == :read && read_permission_action?` both are checking the same?

File: spec/commands/decidim/challenges/admin/create_meeting_spec.rb [hunk 7]
```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
```
Comment (tramuntanal) [inline spec/commands/decidim/challenges/admin/create_meeting_spec.rb:1]: rename file

File: spec/commands/decidim/challenges/admin/update_meeting_spec.rb [hunk 8]
```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
```
Comment (tramuntanal) [inline spec/commands/decidim/challenges/admin/update_meeting_spec.rb:1]: rename file

File: spec/controllers/decidim/challenges/admin/challenges_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+            it "doesn't create a challenge" do
+              post :create, params: params
+
+              expect(flash[:alert]).not_to be_empty
+              expect(response).to have_http_status(:ok)
+            end
+          end
+        end
+
+        describe "DELETE #destroy" do
```
Comment (tramuntanal) [inline spec/controllers/decidim/challenges/admin/challenges_controller_spec.rb:99]: I think we also need the `PUT #update` spec

**Review by tramuntanal — CHANGES_REQUESTED** [review]
Good job @Leusev :clap: !
I've left some comments
Make sure translations to :ca, :en and :es are complete

File: app/assets/config/decidim_challenges_manifest.js (hunk 1 above)
Comment (Leusev) [reply app/assets/config/decidim_challenges_manifest.js:3]: Sure not, I'll remove!

File: app/assets/javascripts/decidim/challenges/add_challenge.js.es6 (hunk 2 above)
Comment (Leusev) [reply app/assets/javascripts/decidim/challenges/add_challenge.js.es6:2]: Nope

**Comment by Leusev** [comment]
All changes done @tramuntanal :smile:

**Review by tramuntanal — APPROVED** [review]

**Pull request merged**